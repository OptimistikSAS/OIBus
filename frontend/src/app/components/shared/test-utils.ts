import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
