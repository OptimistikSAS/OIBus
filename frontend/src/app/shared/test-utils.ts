import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
