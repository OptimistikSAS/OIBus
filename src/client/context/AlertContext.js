import React from 'react'

export const AlertContext = React.createContext({})

export const AlertProvider = AlertContext.Provider
export const AlertConsumer = AlertContext.Consumer
