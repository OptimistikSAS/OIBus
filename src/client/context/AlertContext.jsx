import React from 'react'
import PropTypes from 'prop-types'

const AlertContext = React.createContext({})

const AlertProvider = ({ children }) => {
  const [alert, setAlert] = React.useState({})
  return (
    <>
      <AlertContext.Provider value={{ alert, setAlert }}>
        {children}
      </AlertContext.Provider>
    </>
  )
}
AlertProvider.propTypes = { children: PropTypes.node.isRequired }

export { AlertContext, AlertProvider }
