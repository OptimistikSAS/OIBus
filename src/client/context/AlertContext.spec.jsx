/**
 * @jest-environment jsdom
 */
import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'

import { AlertProvider } from './AlertContext.jsx'

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

describe('AlertProvider', () => {
  test('check AlertProvider', () => {
    act(() => {
      ReactDOM.render(
        <AlertProvider>
          <div />
        </AlertProvider>,
        container,
      )
    })
    expect(container).toMatchSnapshot()
  })
})
