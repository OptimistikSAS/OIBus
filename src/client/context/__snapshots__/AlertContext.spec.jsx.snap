// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logs check AlertProvider 1`] = `<div />`;
